pkg/resource/events-logger-secret/events-logger-secret.go

```diff
@@ -0,0 +1,60 @@
+package eventsloggersecret
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/pkg/errors"
+
+	"github.com/giantswarm/logging-operator/pkg/common"
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	loggingsecret "github.com/giantswarm/logging-operator/pkg/resource/logging-secret"
+)
+
+const eventsLoggerSecretName = "events-logger-secret"
+
+func GenerateEventsLoggerSecret(lc loggedcluster.Interface, loggingCredentialsSecret *v1.Secret, lokiURL string) (v1.Secret, error) {
+	var data map[string][]byte
+	var err error
+
+	switch lc.GetEventsLogger() {
+	case common.EventsLoggerGrafanaAgent:
+		data, err = GenerateGrafanaAgentSecret(lc, loggingCredentialsSecret, lokiURL)
+		if err != nil {
+			return v1.Secret{}, err
+		}
+	case common.EventsLoggerAlloy:
+		// In the case of Alloy being the events logger, we reuse the secret generation from the logging-secret package
+		data, err = loggingsecret.GenerateAlloyLoggingSecret(lc, loggingCredentialsSecret, lokiURL)
```

Comment:
Are we expecting the same secret?

Reply:
In my understanding yes as this secret is only used to authenticate against Loki

Reply:
They share the same format? Because then maybe we can also reuse the same secret and not create a duplicate

Reply:
> They share the same format?

Unfortunately no. At least `promtail`, `grafana-agent` and `alloy` each have a unique format for the secret, hence I only reused the `alloy` secret generation for the events-logger.